pybop/samplers/__init__.py

```diff
@@ -0,0 +1,161 @@
+import logging
+import numpy as np
+from pints import ParallelEvaluator
+from typing import Union
+
+from pybop import LogPosterior
+
+
+class BaseSampler:
+    """
+    Base class for Monte Carlo samplers.
+    """
+
+    def __init__(self, log_pdf, x0, cov0: Union[np.ndarray, float]):
```

Comment:
Update the input from `log_pdf` to `log_posterior` to make it clear that an instance of `LogPosterior` is expected here, not just a likelihood.